### PR TITLE
feat: add more preflight checks

### PR DESCRIPTION
**Pull request by noahpb**
addresses concerns for dubbd in https://github.com/defenseunicorns/uds-cli/issues/4

### CODE REVIEW

File: defense-unicorns-distro/scripts/preflight.sh [hunk 1]
```diff
@@ -15,9 +19,33 @@ supported_minor_version="26"
...
   echo "Server version is not supported"
-  echo "Please update your cluster to a minimum Kubernetes version of ${supported_major_version}.${server_minor_version}"
+  echo "Please update your cluster to a minimum Kubernetes version of ${supported_major_version}.${supported_minor_version}"
   exit 1
 fi
+# system has a kubecontext set
+if [ "$has_kubeconfig" == '""' ]; then 
+  echo "No kubecontext has been set. Please ensure that a valid context is configured before proceeding."
+  exit 1
+fi
```
Comment (mjnagel) [inline defense-unicorns-distro/scripts/preflight.sh:31]: Does it make sense to reverse the order of these two checks? And maybe the health one? Confirm we have a kubeconfig (and maybe confirm cluster health) before checking version?

**Review by MxNxPx — APPROVED** [review]
:ship: :it:

**Review by MxNxPx — CHANGES_REQUESTED** [review]
looking good after you make some of the changes we discussed where the commands run in the component before action always return a 0 exit status and is handled with logic in the script after the fact.

**Review by MxNxPx — DISMISSED** [review]

**Review by mjnagel — APPROVED** [review]
:shipit:

**Pull request merged**